modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/item/action/AnalyticsReportsContentDashboardItemActionProviderImpl.java

```diff
@@ -0,0 +1,217 @@
+import com.liferay.asset.kernel.model.AssetRenderer;
+import com.liferay.asset.kernel.model.AssetRendererFactory;
+import com.liferay.content.dashboard.item.action.ContentDashboardItemAction;
+import com.liferay.content.dashboard.item.action.ContentDashboardItemActionException;
+import com.liferay.info.display.contributor.InfoDisplayContributor;
+import com.liferay.info.display.contributor.InfoDisplayContributorTracker;
+import com.liferay.info.display.contributor.InfoDisplayObjectProvider;
+import com.liferay.layout.page.template.model.LayoutPageTemplateEntry;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.model.Layout;
+import com.liferay.portal.kernel.portlet.LiferayWindowState;
+import com.liferay.portal.kernel.portlet.PortletURLFactory;
+import com.liferay.portal.kernel.security.permission.ActionKeys;
+import com.liferay.portal.kernel.service.ClassNameLocalService;
+import com.liferay.portal.kernel.service.LayoutLocalService;
+import com.liferay.portal.kernel.service.permission.LayoutPermissionUtil;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.ResourceBundleUtil;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import java.util.ResourceBundle;
+
+import javax.portlet.PortletURL;
+import javax.portlet.RenderRequest;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author David Arques
+ */
+@Component(service = AnalyticsReportsContentDashboardItemActionProvider.class)
+public class AnalyticsReportsContentDashboardItemActionProviderImpl
+	implements AnalyticsReportsContentDashboardItemActionProvider {
+
+	@Override
+	public ContentDashboardItemAction getContentDashboardItemAction(
+			String className, long classPK,
+			HttpServletRequest httpServletRequest)
+		throws ContentDashboardItemActionException {
+
+		try {
+			if (!isShowContentDashboardItemAction(
+					className, classPK, httpServletRequest)) {
+
+				return null;
+			}
+
+			ResourceBundle resourceBundle = ResourceBundleUtil.getBundle(
+				_portal.getLocale(httpServletRequest), getClass());
+
+			PortletURL portletURL = _portletURLFactory.create(
+				httpServletRequest,
+				AnalyticsReportsPortletKeys.ANALYTICS_REPORTS,
+				RenderRequest.RENDER_PHASE);
+
+			portletURL.setParameter("mvcPath", "/analytics_reports_panel.jsp");
+
+			portletURL.setWindowState(LiferayWindowState.EXCLUSIVE);
+
+			portletURL.setParameter(
+				"classNameId",
+				String.valueOf(
+					_classNameLocalService.getClassNameId(className)));
+
+			portletURL.setParameter("classPK", String.valueOf(classPK));
+
+			return new AnalyticsReportsContentDashboardItemAction(
+				_language.get(resourceBundle, "view-metrics"),
```

Comment:
Could be better to init AnalyticsReportsContentDashboardItemAction with a resourceBundle and return the label later?

Reply:
I'm not sure I got it, you mean, initializing just with the label key?

Reply:
I mean, the actions should know its name, right?

Reply:
ah, I got it now, so, passing language and the resourcebundle in the constructor so the label could be properly initiated

Reply:
I passed the ResourceBundle, and getLabel remains the same, without Locale since it was already set in the resource bundle